app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -492,6 +492,12 @@ class HomeFragment : Fragment() {
             track(Event.HomeScreenViewCount)
         }
 
+        consumeFrom(appStore) {
+            if (shouldEnableWallpaper()) {
+                it.wallpaper.applyToView(binding.homeLayout)
+            }
+        }
+
```

Comment:
@gabrielluong mentioned that `consumeFrom` should be used over `Store.observe(viewLifecycleOwner)`. It looks like there are some [comments](https://github.com/mozilla-mobile/android-components/blob/379cb84d8874da4e3b7629f370f745ec2d4b1ec0/components/lib/state/src/main/java/mozilla/components/lib/state/ext/Fragment.kt#L38) describing a bug in viewLifecycleOwner that require this.

@pocmo started a discussion about it in [this issue](https://github.com/mozilla-mobile/android-components/issues/3828). It's been quite a while since this issue was opened, and I wonder if it's still applicable. We are recreating the activity to change the theme when the wallpaper changes, and I saw no issues when my implementation used

```
appStore.observe {
    lifecycleScope.launch(Main) {
    ...
    }
}

Reply:
Now that I am thinking about this I think we should probably look at wrapping this in a Feature such as `TopSitesFeature` or `RecentTabs`. Ideally, we don't have too much logic inside of `HomeFragment`.

Reply:
Could we do this in follow up? It will be deal to start testing the settings page in nightly as soon as possible.

Reply:
I guess that's fine, but I would maybe recommend landing this after the cut.